### PR TITLE
Config refactor:  switch to dynaconf and add the 'LLM_VM' prefix to env vars

### PR DESCRIPTION
**Pull request by dorkitude**
- Dynaconf is working splendidly.  I removed our hardcoded defaults and put them into a file called settings.default.toml.   Users can provide their own settings.toml which will take precedence over the defaults.

- Environment variables (including your .env file) take precedence over both the TOML files.   To be good citizens, we need to start to namespace our environment variables.   For instance, my laptop has like 6 open AI keys across projects, and one of those projects also uses "OPENAI_API_KEY" without a prefix.  This can create very bad news for users.  I've opted for the prefix LLM_VM, so once we merge this you'll need to update your environment variables (or .env file if you use that) to have names like LLM_VM_OPENAI_API_KEY

- if none of these files are present, dynaconf will look in the XDG standard location (I think it's ~/.config/projectname, which is clean since we're also planning to follow the XDG standards for ~/.cache/projectname...  for temp files and ~/.local/share/projectname... for e.g. SQLite storage and flat storage)

### CODE REVIEW

**Comment by dorkitude** [comment]
Closes #79

File: src/llm_vm/onsite_llm.py [hunk 1]
```diff
@@ -167,17 +212,18 @@ class Small_Local_Pythia(Base_Onsite_LLM):
...
+    #     super().__init__(kwargs) ## this line is required
+    model_uri = "EleutherAI/pythia-70m-deduped"
     def model_loader(self):
         return GPTNeoXForCausalLM.from_pretrained(self.model_uri)
     def tokenizer_loader(self):
         return AutoTokenizer.from_pretrained(self.model_uri)
 
-  
+
+>>>>>>> main
```
Comment (cartazio) [inline src/llm_vm/onsite_llm.py:224]: doh!

File: src/llm_vm/onsite_llm.py [hunk 2]
```diff
@@ -133,16 +175,19 @@ def asynctune():
...
             new_model = os.path.join(model_path_default,"finetuned_models",self.model_name, timestamp + '_' + self.model_name + ".pt" )
             open(new_model,"a")
             torch.save(self.model.state_dict(), new_model) # the model in memory is different now
-            self.model_name = self.model_name + "_ft_"+  timestamp 
+            self.model_name = self.model_name + "_ft_"+  timestamp
             optimizer.storage.set_model(c_id, new_model)
             return math.exp(eval_results['eval_loss']) #perplexity is the metric we use for finetuning measurement
         return asynctune
-    
+<<<<<<< HEAD
```
Comment (cartazio) [inline src/llm_vm/onsite_llm.py:187]: merge conflict unresolved?

File: src/llm_vm/onsite_llm.py [hunk 3]
```diff
@@ -85,17 +122,22 @@ def generate(self,prompt,max_length=100,**kwargs): # both tokenizer and model ta
...
         inputs=self.tokenizer(prompt,return_tensors="pt")
         generate_ids=self.model.generate(inputs.input_ids,max_length=max_length)
         resp= self.tokenizer.batch_decode(generate_ids,skip_special_tokens=True,clean_up_tokenization_spaces=False)[0]
-        # need to drop the len(prompt) prefix with these sequences generally 
-        # because they include the prompt. 
-        return resp[len(prompt):] 
+<<<<<<< HEAD
+        # need to drop the len(prompt) prefix with these sequences generally
+        return resp[len(prompt):]
+=======
```
Comment (cartazio) [inline src/llm_vm/onsite_llm.py:136]: bad merge?

File: src/llm_vm/onsite_llm.py [hunk 4]
```diff
@@ -73,7 +74,43 @@ def model_loader(self):
...
+        self.tokenizer=self.tokenizer_loader()
+        self.model= self.model_loader()
+
+    def model_loader(self):
+        return GPTNeoXForCausalLM.from_pretrained(self.model_uri)
+    def tokenizer_loader(self):
+        return AutoTokenizer.from_pretrained(self.model_uri)
+=======
+
+>>>>>>> main
```
Comment (cartazio) [inline src/llm_vm/onsite_llm.py:113]: bad merge?

File: src/llm_vm/config.py [hunk 5]
```diff
@@ -1,27 +1,50 @@
...
 
-# Set parse strategy here, lines should be self explanatory
-parser.add_argument("-b", "--big_model", type=str, default='chat_gpt', help='Big LLM Model. Default: chat_gpt')
-parser.add_argument("-c", "--config_file", type=str, help='Config File')
-parser.add_argument("-p", "--port", type=int, default=3002, help='Port Number. Default: 3002')
-parser.add_argument("-s", "--small_model", type=str, default='bloom', help='Small LLM Model. Default: bloom')
-parser.add_argument("-H", "--host", type=str, default='127.0.0.1', help='Host Address. Default: 127.0.0.1')
+# Set the CLI argument values to environment variables if they are present
+if args.big_model is not None:
+    os.environ['LLM_VM_BIG_MODEL'] = args.big_model
```
Comment (cartazio) [inline src/llm_vm/config.py:18]: wait, is this setting the Process Env Variables?

**Comment by cartazio** [comment]
can you rebase again? (andrew had to fix a blocking test failure)

File: src/llm_vm/onsite_llm.py (hunk 3 above)
Comment (dorkitude) [reply src/llm_vm/onsite_llm.py]: fixed all these, thanks!

**Comment by dorkitude** [comment]
w00t all the tests run!  merging to main

**Pull request merged**